fork_choice_control/src/mutator.rs

```diff
@@ -1269,6 +1279,26 @@ where
         }
     }
 
+    fn handle_finish_persisting_data_column_sidecars(
+        &mut self,
+        wait_group: W,
+        persisted_data_column_ids: Vec<DataColumnIdentifier>,
+    ) {
+        self.store_mut().mark_persisted_data_columns(persisted_data_column_ids);
+
+        self.update_store_snapshot();
+
+        if self.store.has_unpersisted_data_column_sidecars() {
+            self.spawn(PersistDataColumnSidecarsTask {
+                store_snapshot: self.owned_store(),
+                storage: self.storage.clone_arc(),
+                mutator_tx: self.owned_mutator_tx(),
+                wait_group,
+                metrics: self.metrics.clone(),
+            });
+        }
+    }
+
```

Comment:
This implementation looks solid to me. However, while it's unrelated to your PR, I feel compelled to comment on the struct as a whole.

Is a 2500+ line struct the best approach here? It seems a bit unwieldy, and to be honest, it's raising some red flags for me in terms of maintainability. A struct isn't just a namespace, and using it in this way feels like it might be leaning too heavily on object-oriented design principles, which aren't always a natural fit in Rust.

Perhaps it's worth considering whether some of the interior mutability could be separated out from the rest of the functionality. Refactoring parts of this struct into smaller, more focused modules or components could not only make the codebase cleaner and easier to maintain but also improve performance and reduce the potential for bugs.

What do you think? Is this something we should explore further and consider raising a GitHub issue on?